### ISSUE TITLE
Duplicate queries not correctly detected

### ISSUE THREAD

**Issue by rubengrill** [jazzband/django-debug-toolbar#1239]
With django debug toolbar 2.2 (since 2.0) I see a lot of duplicated queries.
When turning on logging for sql queries, I see them only once, and when debugging, those queries are clearly executed only once.

I found out that this only happens when using chunked cursors, because debug toolbar wraps the cursor twice, leading to logging it twice and wrongly marking it as duplicated.

Please see this line:
https://github.com/jazzband/django-debug-toolbar/blob/2.2/debug_toolbar/panels/sql/tracking.py#L58

It wraps the cursor returned from `connection._djdt_chunked_cursor`.
For databases without a special implementation for chunked cursors (e.g. mysql), that method actually just runs `connection.cursor`:
https://github.com/django/django/blob/2.2.10/django/db/backends/base/base.py#L572

Which in turn wraps the cursor a second time:
https://github.com/jazzband/django-debug-toolbar/blob/2.2/debug_toolbar/panels/sql/tracking.py#L53

**Comment by tim-schilling** [comment]
Do you have a test case that we can reproduce this with?

**Comment by valentijnscholten** [comment]
I am seeing the same, not sure how I can easily create a test case as I am seeing this in a project with a `models.py` which by itself is over 3000 lines...

**Comment by carltongibson** [comment]
I just hit this with SQLite. I'll try and follow-up with a sample project next week, but it's just simple model. 

Commenting out the overriding of the `chunked_cursor` (as suggested per @rubengrill's https://github.com/jazzband/django-debug-toolbar/issues/1239#issue-573597111) revolves the issue.

I'm guessing a check to see if the returned cursor is already wrapped might be sufficient. 🤔

**Comment by saemideluxe** [comment]
I encountered the same problem while trying to investigate a performance issue. This is rather confusing because one sees the same query and the same stack trace but different times. Preventing the wrapping of chunked_cursor resolved the issue. I made a PR #1475 but not sure if this is acceptable like that.
Btw. I stumbled over your talk about ModelChoiceField @carltongibson due to the same issue. Thanks, good and helpfull talk :)
